app/Http/Controllers/FileController.php

```diff
@@ -96,8 +99,55 @@ public function store(Request $request)
             
             // $error_data = $this->checkValues($uploader_id);
 
+
+            // check number of uploaded records
+            $sqlUploadedRecords = " SELECT COUNT(*) as number_of_records FROM data_template WHERE uploader_id = '" . $uploader_id . "';";
+
+            // execute custom SELECT SQL
+            $uploadedRecordsResults = DB::select($sqlUploadedRecords);
+            $numberUploadedRecords = $uploadedRecordsResults[0]->number_of_records;
+
+            
+            // check number of records already existed in database
+            $sqlExistedRecords  = " SELECT COUNT(*) as number_of_records";
+            $sqlExistedRecords .= " FROM data ta, data_template tb";
+            $sqlExistedRecords .= " WHERE tb.uploader_id = '" . $uploader_id . "'";
+            $sqlExistedRecords .= " AND ta.fecha_hora = tb.fecha_hora";
+            $sqlExistedRecords .= " AND ta.id_station = tb.id_station;";
+
+            // execute custom SELECT SQL
+            $existedRecordsResults = DB::select($sqlExistedRecords);
+            $numberExistedRecords = $existedRecordsResults[0]->number_of_records;
+
+            
+            // number of not existed records = number of uploaded records - number of existed records
+            $numberNotExistedRecords = $numberUploadedRecords - $numberExistedRecords;
+
+
+
+            // prepare advice message
+            $adviceMessage = "";
+            $flagUploadable = 0;
+
+            if ($numberNotExistedRecords == $numberUploadedRecords) {
+                $adviceMessage = "All " . $numberUploadedRecords . " record(s) are new records. Please kindly confirm to upload this data file.";
+                $flagUploadable = 1;
+            } else if ($numberExistedRecords == $numberUploadedRecords) {
+                $adviceMessage = "All " . $numberExistedRecords .  " record(s) are already existed in system. Please kindly cancel this upload.";
+                $flagUploadable = 0;
+            } else {
+                $adviceMessage = $numberExistedRecords . " out of " . $numberUploadedRecords . " records are already existed in system. Please kindly cancel this upload and further check data file correctness.";
+                $flagUploadable = 0;
```

Comment:
I think we need an option to upload anyway even when some records already exist. For example when 2 files overlap in dates - we still need to upload the 2nd file to get the extra records. But it's worth highlighting this message to the user, and maybe ask them to tick a 'confirm' box before the final submit + save button?

Reply:
Hi Dave,

Thank you very much for your help.

I just tested and here are some feedback:

#### 0B: Call python or Pipenv depending on environment
  - I have added "PIPENV=false" to my .env file. It works very well, thanks a lot
 
#### 9B - Disable submit button if it should not upload
  - When there are records found in database, "Confirm" button is disabled properly as expected
  - Carlos will ask Bolivia team to suggest whether we should disable "Confirm" button when some records existed in database
  - It would be nice to let user to decide to continue to upload or further check and come back later
  - Currently we execute one very big INSERT SQL to insert all records. 
    If SQL error occured due to duplication of any records, all records will not be inserted into database.
    We will need to modify program to execute one INSERT SQL for each record individially with error handling.
   
#### 14: Divert user to another page after clicking confirm button, or disable confirm button
  - When there is no records existed in database, "Confirm" button is enabled, after clicking it, it divert to a page that not existed. 404 Page Not Found error showed.
  - TODO: add a static page with success message and options:
   - Go back to home page
   - Upload another file
 
#### 13D - replace hard-coded img paths for station confirmation popup
  - Station images should be collected one time when adding a new station.
  - Suppose it is not likely to update station images after initial setup.
  - As the objective is to provide visual identification for selecting a correct station,
    probably it is a good idea to disallow user to upload images by themselves
  - Confirmation between Bolivia team and Stats4SD is required to ensure correctness of station images.
  - We can prepare dummy images for all stations and ask Bolivia team for real images later on
  
  - Perhaps we can go for a simpler approach:
   - remove 3 new columns in table
   - prepare dummy images for all stations

Reply:
> I have added "PIPENV=false" to my .env file. It works very well, thanks a lot

great!

> It would be nice to let user to decide to continue to upload or further check and come back later

Agreed

> We will need to modify program to execute one INSERT SQL for each record individially with error handling.

As long as that remains efficient and won’t cause issues for the user later. If some records return errors, we would need to show those errors to the user (or a summary like “900 records entered successfully, 5 records caused errors”. In these cases, would the suggestion be for the user to review the data in those records, fix the data and then try the file upload again? (At which point it will show 900 dups, and 5 new records). - I think that sounds good to me.

> Perhaps we can go for a simpler approach:
> remove 3 new columns in table
> prepare dummy images for all stations

Sounds good to me for now.